music_assistant/providers/bandcamp/__init__.py

```diff
@@ -0,0 +1,404 @@
+"""Bandcamp music provider support for MusicAssistant."""
+
+import asyncio
+from collections.abc import AsyncGenerator
+from contextlib import suppress
+
+from bandcamp_async_api import (
+    BandcampAPIClient,
+    BandcampAPIError,
+    BandcampNotFoundError,
+    SearchResultAlbum,
+    SearchResultArtist,
+    SearchResultTrack,
+)
+from bandcamp_async_api.models import CollectionType
+from music_assistant_models.config_entries import ConfigEntry, ConfigValueType, ProviderConfig
+from music_assistant_models.enums import (
+    ConfigEntryType,
+    ContentType,
+    MediaType,
+    ProviderFeature,
+    StreamType,
+)
+from music_assistant_models.errors import MediaNotFoundError
+
+# noinspection PyProtectedMember
+from music_assistant_models.media_items import (
+    Album,
+    Artist,
+    AudioFormat,
+    SearchResults,
+    Track,
+)
+from music_assistant_models.provider import ProviderManifest
+from music_assistant_models.streamdetails import StreamDetails
+
+from music_assistant.controllers.cache import use_cache
+from music_assistant.helpers.throttle_retry import ThrottlerManager, throttle_with_retries
+from music_assistant.mass import MusicAssistant
+from music_assistant.models import ProviderInstanceType
+from music_assistant.models.music_provider import MusicProvider
+
+from .converters import BandcampConverters
+
+SUPPORTED_FEATURES = {
+    ProviderFeature.LIBRARY_ARTISTS,
+    ProviderFeature.LIBRARY_ALBUMS,
+    ProviderFeature.LIBRARY_TRACKS,
+    # ProviderFeature.BROWSE,  # TODO: Consider
+    ProviderFeature.SEARCH,
+    ProviderFeature.ARTIST_ALBUMS,
+    ProviderFeature.ARTIST_TOPTRACKS,
+    # ProviderFeature.RECOMMENDATIONS,  # TODO: Consider
+    # ProviderFeature.SIMILAR_TRACKS,  # TODO: Consider
```

Comment:
Could you use this for a "Don't stop the music" feature?  Just keep taking the first "similar track" that hasn't been played, storing the remainder (up to some limit)?

Recommendations - I guess you _could_ treat these as another "Don't stop the music" source and randomly pick one, if they're available, instead of a "similar track".